src/variant.cpp

```diff
@@ -2064,6 +2064,17 @@ Variant* Variant::conclude() {
         connect_directions.push_back(SOUTH_EAST);
     }
 
+    // If not a connect variant, set connectPieceTypes to no pieces.
+    if ( !(connectRegion1[WHITE] || connectRegion1[BLACK] || connectN || connectNxN || collinearN) )
+    {
+          connectPieceTypes = NO_PIECE_SET;
+    }
+    //Otherwise optimize to pieces actually in the game.
+    else if (connectPieceTypes == CHESS_PIECES)
```

Comment:
I think it is more generic to simply have
```
else
    connectPieceTypes = connectPieceTypes & pieceTypes;
```
This way you just ignore what is not used and keep what is used.